### ISSUE TITLE
dex.ksy: wrong license, should be CC-BY-3.0 or CC-BY-4.0

### ISSUE THREAD

**Issue by armijnhemel** [kaitai-io/kaitai_struct_formats#508]
the file dex.ksy currently says:

```
  # Portions of this page are reproduced from work created and shared by Google and used
  # according to terms described in the Creative Commons 3.0 Attribution License.
```

and then the license field says:

```
 license: CC-BY-SA-3.0
```

but these are not the same licenses. CC-BY-SA-3.0 is "Creative Commons 3.0 Attribution-ShareAlike 3.0", which is actually a different license. The correct license is CC-BY-3.0.

By the way, it seems that Google has updated the license of their content to CC-BY-4.0: https://source.android.com/license (although both 3.0 and 4.0 are mentioned, I will try to clarify that with Google).

**Comment by generalmimon** [comment]
> but these are not the same licenses. CC-BY-SA-3.0 is "Creative Commons 3.0 Attribution-ShareAlike 3.0", which is actually a different license.

You're right. This turns out to be a suggestion by @GreyCat in 2017 (https://github.com/kaitai-io/kaitai_struct_formats/commit/2f20df5551dbea39b12a087d8d06b807140f5883#r22802873) on the commit adding the `dex.ksy` file by @koczkatamas, but unfortunately it's not well thought out. The quotations in his comment were apparently taken from https://developers.google.com/terms/site-policies, but the problem is that this page states licensing of the content on Google Developers (the domain *developers.google.com*), not on the AOSP site at **source.android.com** where the https://source.android.com/devices/tech/dalvik/dex-format (from `doc-ref` key) resides at.

The content license for the _source.android.com_ is stated on the page [Content License](https://web.archive.org/web/20170710084346/https://source.android.com/license). https://source.android.com/devices/tech/dalvik/dex-format is clearly a documentation page, so we should infer the license for _documentation content_:

> # Content License
> 
> For the purposes of licensing, the content of this web site is divided into two categories:
> 
> * Documentation content, including both static documentation and content extracted from source code modules, as well as sample code, and
> * All other site content
> 
> The documentation content on this site is made available to you as part of the [Android Open Source Project](https://web.archive.org/web/20170710084346/https://android.googlesource.com/). **This documentation, including any code shown in it, is licensed under the [Apache 2.0 license](https://web.archive.org/web/20170710084346/https://www.apache.org/licenses/LICENSE-2.0),** the preferred license for all parts of the of the Android Open Source Project.

Also see https://web.archive.org/web/20170710084346/https://source.android.com/license#attribution:

> ## Attribution
> 
> Proper attribution is required when you reuse or create modified versions of content that appears on a page made available under the terms of the Creative Commons Attribution license. **On this site, the requirement for attribution applies only to the non-documentation content, as described earlier in this document.**

... so no attribution is required in our case.

In short, I think it should've been the license [`Apache-2.0`](https://spdx.org/licenses/Apache-2.0.html) in `/meta/license` all along (instead of any Creative Commons license), with no attribution like _"Portions of this page are reproduced from work created and shared by Google and (...)"_ required.

**Comment by armijnhemel** [comment]
Ah, yes, you are right, it seems I also misread (but at least I was correct about CC-BY-SA-3.0 being incorrect ;-) )

**Comment by generalmimon** [comment]
> In short, I think it should've been the license [`Apache-2.0`](https://spdx.org/licenses/Apache-2.0.html) in `/meta/license` all along (instead of any Creative Commons license), with no attribution like _"Portions of this page are reproduced from work created and shared by Google and (...)"_ required.

Hmm, should we actually reach out to all `dex.ksy` contributors again when we want to change the license? I usually do this when `/meta/license` is missing entirely (for example https://github.com/kaitai-io/kaitai_struct_formats/issues/411) because that is the case when all contributors hold exclusive copyright to their changes, but I guess that this is needed if we want to change the license too. Strictly speaking, we probably need to ensure that people who made the changes under the license `CC-BY-SA-3.0` would be OK with releasing them under `Apache-2.0` instead.

**Comment by armijnhemel** [comment]
It would be the proper thing to do. It seems that one contributor hasn't been active for quite some time though. A possible solution: set a hard deadline (a month, two months), send notifications, and replace contributions from that developer if there is no response.

**Comment by koczkatamas** [comment]
Hello everyone. I just wanted to pre-confirm that I am okay with changing the license from CC to Apache.

**Comment by generalmimon** [comment]
I'll put a list of contributors here according to [Git history](https://github.com/kaitai-io/kaitai_struct_formats/commits/434801a79010aef4fb4a72dedcbb2ba0012bb7bf/executable/dex.ksy). If you're mentioned, **please comment** whether you are OK with relicensing your changes in the [`executable/dex.ksy`](https://github.com/kaitai-io/kaitai_struct_formats/blob/434801a79010aef4fb4a72dedcbb2ba0012bb7bf/executable/dex.ksy) spec from [<del>`CC-BY-SA-3.0`</del>](https://spdx.org/licenses/CC-BY-SA-3.0.html) to **[`Apache-2.0`](https://spdx.org/licenses/Apache-2.0.html)**.

* [`executable/dex.ksy`](https://github.com/kaitai-io/kaitai_struct_formats/commits/434801a79010aef4fb4a72dedcbb2ba0012bb7bf/executable/dex.ksy)

  - [x] @koczkatamas ([commits](https://github.com/kaitai-io/kaitai_struct_formats/commits/434801a79010aef4fb4a72dedcbb2ba0012bb7bf/executable/dex.ksy?author=koczkatamas))
  - [ ] @GreyCat ([commits](https://github.com/kaitai-io/kaitai_struct_formats/commits/434801a79010aef4fb4a72dedcbb2ba0012bb7bf/executable/dex.ksy?author=GreyCat))
  - [ ] @silviupopescu ([commits](https://github.com/kaitai-io/kaitai_struct_formats/commits/434801a79010aef4fb4a72dedcbb2ba0012bb7bf/executable/dex.ksy?author=silviupopescu))
  - [ ] @generalmimon ([commits](https://github.com/kaitai-io/kaitai_struct_formats/commits/434801a79010aef4fb4a72dedcbb2ba0012bb7bf/executable/dex.ksy?author=generalmimon))

**Comment by silviupopescu** [comment]
Hi, I'm OK with changing the license.

**Comment by generalmimon** [comment]
I hereby agree to relicense [my changes](https://github.com/kaitai-io/kaitai_struct_formats/commits/434801a79010aef4fb4a72dedcbb2ba0012bb7bf/executable/dex.ksy?author=generalmimon) in [`executable/dex.ksy`](https://github.com/kaitai-io/kaitai_struct_formats/blob/434801a79010aef4fb4a72dedcbb2ba0012bb7bf/executable/dex.ksy) to the [`Apache-2.0`](https://spdx.org/licenses/Apache-2.0.html) license.

**Comment by generalmimon** [comment]
@GreyCat, can you please also comment? I would like to solve this quickly.

**Comment by GreyCat** [comment]
I hereby also agree to relicense [my changes](https://github.com/kaitai-io/kaitai_struct_formats/commits/434801a79010aef4fb4a72dedcbb2ba0012bb7bf/executable/dex.ksy?author=greycat) in [`executable/dex.ksy`](https://github.com/kaitai-io/kaitai_struct_formats/blob/434801a79010aef4fb4a72dedcbb2ba0012bb7bf/executable/dex.ksy) to the [`Apache-2.0`](https://spdx.org/licenses/Apache-2.0.html) license.

**Comment by generalmimon** [comment]
Thanks to everyone involved in resolving this issue!
